dewy/tasks/dewy_tasks.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+from typing import Any, Coroutine
+from urllib.parse import urlparse
+
+import asyncpg
+import taskiq
+from taskiq.serializers import CBORSerializer
+
+from dewy.config import ServeConfig
+
+from ._ingest import ingest_task
+
+
+class _KeywordMiddleware(taskiq.TaskiqMiddleware):
+    """Midleware to make sure all calls use keyword arguments.
+
+    This avoids a lot of issues with changing task signatures."""
+
+    def pre_send(
+        self, message: taskiq.TaskiqMessage
+    ) -> taskiq.TaskiqMessage | Coroutine[Any, Any, taskiq.TaskiqMessage]:
+        assert (
+            len(message.args) == 0
+        ), f"All arguments should be passed by keyword, got: {message.args}"
+        return super().pre_send(message)
+
+
+def _create_broker(config: ServeConfig) -> taskiq.AsyncBroker:
+    if config.broker is None:
+        return taskiq.InMemoryBroker()
+    else:
+        broker_url = urlparse(config.broker)
+        if broker_url.scheme == "amqp":
+            from taskiq_aio_pika import AioPikaBroker
+
+            return AioPikaBroker(url=config.broker)
+        else:
+            raise ValueError(
+                f"Unsupported scheme '{broker_url.scheme}' in broker URL: '{config.broker}'"
+            )
+
+
+class DewyTasks:
+    """
+    The main class providing the dewy worker.
+    """
+
+    def __init__(self, pg_pool: asyncpg.Pool, config: ServeConfig) -> None:
+        broker = (
+            _create_broker(config)
+            .with_middlewares(_KeywordMiddleware())
+            .with_serializer(CBORSerializer())
```

Comment:
CBOR is necessary instead of a JSON format (which would make the messages easier to print / debug) because when we ingest uploaded content, the bytes are sent as part of the message.

This could be improved upon:

1. We could store the content to S3.
2. We could store the content to local disk and maybe delete it after ingestion completes (if we didn't want to support re-extraction).
3. We could store the content in the DB.

I'm not sure we want to do any of these changes now, but it may be worth considering when we start needing to debug ingestion.